#VRML_SIM R2022a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.7248818625560527 -0.6887544410957545 -0.012790825154892367 3.1820785599099652
  position -0.03921118749131748 10.585953475702988 0.730199969013195
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF myrobot Robot {
  translation -2.091775025995208 0.06913150340593657 2.155852692458413
  rotation -0.006976195902161351 0.5899819573026123 -0.8073862909091989 0.010171461148811498
  children [
    GPS {
      children [
        Solid {
          rotation 0.014477377716948797 0.0036947172358512783 -0.999888371068985 -0.0002553109912272422
          children [
            DEF gps_shape Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
              }
              geometry Sphere {
                radius 0.01
              }
            }
          ]
          name "gps_solid"
          boundingObject USE gps_shape
          physics Physics {
          }
        }
      ]
      name "world_gps"
      boundingObject USE gps_shape
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0 0.01 0.05
      rotation 0 1 0 -1.57
      children [
        DEF ddd Shape {
          appearance PBRAppearance {
            baseColor 0.333333 1 0.498039
          }
          geometry Sphere {
            radius 0.01
          }
        }
      ]
      name "distance_sensor_r"
      boundingObject USE ddd
      physics Physics {
      }
      lookupTable [
        0 10 0
        10 0 0
      ]
    }
    DistanceSensor {
      translation 0 0.01 -0.05
      rotation 0 1 0 1.57
      children [
        DEF ddd Shape {
          appearance PBRAppearance {
            baseColor 0.333333 1 0.498039
          }
          geometry Sphere {
            radius 0.01
          }
        }
      ]
      name "distance_sensor_l"
      boundingObject USE ddd
      physics Physics {
      }
      lookupTable [
        0 10 0
        10 0 0
      ]
    }
    DistanceSensor {
      translation 0.1 0.01 0
      rotation 1 0 0 1.57
      children [
        DEF ddd Shape {
          appearance PBRAppearance {
            baseColor 0.333333 1 0.498039
          }
          geometry Sphere {
            radius 0.01
          }
        }
      ]
      name "distance_sensor"
      boundingObject USE ddd
      physics Physics {
      }
      lookupTable [
        0 10 0
        10 0 0
      ]
    }
    RobotisLds01 {
      translation 0 0.05 0
      rotation -0.9999998414659299 0.00039816335536657546 0.0003981633553666367 1.5707964853289793
      noise 0.0001
    }
    BallJoint {
      jointParameters BallJointParameters {
        position -488.0356033989992
        anchor 0.07 -0.04 0
      }
      jointParameters2 JointParameters {
        position -684.793785573184
      }
      jointParameters3 JointParameters {
        position 4142.648987412737
      }
      endPoint Solid {
        translation 0.07002069323443605 -0.03988849945441233 -1.9660000652788057e-05
        rotation -0.6620980549432166 -0.5000095382000156 0.5582263226948515 2.4650595970820097
        children [
          DEF ball Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
            }
            geometry Sphere {
              radius 0.03
            }
          }
        ]
        name "solid(2)"
        boundingObject USE ball
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 48369.11756071528
        axis 0 0 1
        anchor -0.06 -0.04 -0.05
      }
      device [
        PositionSensor {
          name "l_p_s"
        }
        RotationalMotor {
          name "left_motor"
          acceleration 20
          controlPID 10 4 0
          maxVelocity 100
          minPosition -10
          maxPosition 100000000000
        }
      ]
      endPoint Solid {
        translation -0.06001190087682094 -0.03999676776967329 -0.04999753404665687
        rotation 0 0 1 4.669356045357543
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.03
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 27162.141278379226
        axis 0 0 1
        anchor -0.06 -0.04 0.05
      }
      device [
        PositionSensor {
          name "r_p_s"
        }
        RotationalMotor {
          name "right_motor"
          acceleration 20
          controlPID 10 4 0
          maxVelocity 100
          minPosition -10
          maxPosition 100000000000
        }
      ]
      endPoint Solid {
        translation -0.060005846674450775 -0.03999803482409784 0.05000637678782845
        rotation 0 0 1 0.5990802447330777
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.03
            }
          }
        ]
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF body Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.2 0.07 0.1
      }
    }
  ]
  name "myrobot"
  boundingObject USE body
  physics DEF body Physics {
  }
  controller "ab_avoid"
  supervisor TRUE
}
RectangleArena {
  rotation -0.5773502691896258 0.5773502691896258 0.5773502691896258 2.0943951023931957
  floorSize 5 5
  wallHeight 0.5
}
Solid {
  translation -0.68 0.24383618254126713 -2.0392212016928518e-07
  rotation -1 -1.8643334308650607e-16 1.5634716797766022e-16 1.5707963267911547
  children [
    DEF OB Shape {
      appearance Appearance {
      }
      geometry Cylinder {
        height 0.5
        radius 0.1
      }
    }
  ]
  boundingObject USE OB
  physics Physics {
  }
}
Solid {
  translation -1.64 0.24383618254126693 0.6599999999938799
  rotation -1 -6.699154154082947e-17 1.4561319847095071e-16 1.5707963267911547
  children [
    DEF OB Shape {
      appearance Appearance {
      }
      geometry Cylinder {
        height 0.5
        radius 0.1
      }
    }
  ]
  name "solid(1)"
  boundingObject USE OB
  physics Physics {
  }
}
Solid {
  translation 0.6 0.24383618254126738 -0.2800000000061203
  rotation -1 -1.5473804659016304e-16 2.042042727240156e-16 1.5707963267911547
  children [
    DEF OB Shape {
      appearance Appearance {
      }
      geometry Cylinder {
        height 0.5
        radius 0.1
      }
    }
  ]
  name "solid(2)"
  boundingObject USE OB
  physics Physics {
  }
}
Solid {
  translation -0.41 0.2438361825412672 -1.42000000000612
  rotation -1 -1.6256049991903021e-16 1.523883633748832e-16 1.5707963267911547
  children [
    DEF OB Shape {
      appearance Appearance {
      }
      geometry Cylinder {
        height 0.5
        radius 0.1
      }
    }
  ]
  name "solid(3)"
  boundingObject USE OB
  physics Physics {
  }
}
Solid {
  translation -1.41 0.243836182541267 -0.9600000000061201
  rotation -1 -9.543859792048772e-17 1.1578559644544185e-16 1.5707963267911547
  children [
    DEF OB Shape {
      appearance Appearance {
      }
      geometry Cylinder {
        height 0.5
        radius 0.1
      }
    }
  ]
  name "solid(6)"
  boundingObject USE OB
  physics Physics {
  }
}
Solid {
  translation 1.01 0.24383618254126746 -1.61000000000612
  rotation -1 -1.1133317723156256e-16 1.6614884367335552e-16 1.5707963267911549
  children [
    DEF OB Shape {
      appearance Appearance {
      }
      geometry Cylinder {
        height 0.5
        radius 0.1
      }
    }
  ]
  name "solid(7)"
  boundingObject USE OB
  physics Physics {
  }
}
Solid {
  translation 1.77 0.24383618254126763 0.5199999999938799
  rotation -1 -1.0050362412303317e-16 1.0827178978374892e-16 1.5707963267911547
  children [
    DEF OB Shape {
      appearance Appearance {
      }
      geometry Cylinder {
        height 0.5
        radius 0.1
      }
    }
  ]
  name "solid(8)"
  boundingObject USE OB
  physics Physics {
  }
}
Solid {
  translation -0.53 0.24383618254126718 1.07999999999388
  rotation -1 -1.462773147675094e-16 2.18978166786729e-16 1.5707963267911547
  children [
    DEF OB Shape {
      appearance Appearance {
      }
      geometry Cylinder {
        height 0.5
        radius 0.1
      }
    }
  ]
  name "solid(9)"
  boundingObject USE OB
  physics Physics {
  }
}
Solid {
  translation 1.67 0.2438361825412676 -0.5600000000061202
  rotation -1 -6.758425257912581e-17 1.133632446755967e-16 1.5707963267911547
  children [
    DEF OB Shape {
      appearance Appearance {
      }
      geometry Cylinder {
        height 0.5
        radius 0.1
      }
    }
  ]
  name "solid(4)"
  boundingObject USE OB
  physics Physics {
  }
}
Solid {
  translation -1.279615255430793e-16 0.24383618254126727 1.83999999999388
  rotation -1 -1.8061200146522916e-16 1.6038299878580446e-16 1.5707963267911549
  children [
    DEF OB Shape {
      appearance Appearance {
      }
      geometry Cylinder {
        height 0.5
        radius 0.1
      }
    }
  ]
  name "solid(5)"
  boundingObject USE OB
  physics Physics {
  }
}
Solid {
  translation 1.08 0.24383618254126746 1.35999999999388
  rotation -1 -1.3159292019951187e-16 1.294830576104834e-16 1.5707963267911547
  children [
    DEF OB Shape {
      appearance Appearance {
      }
      geometry Cylinder {
        height 0.5
        radius 0.1
      }
    }
  ]
  name "solid(10)"
  boundingObject USE OB
  physics Physics {
  }
}
